lib/types/index.d.ts

```diff
@@ -115,12 +130,19 @@ declare module 'react-native-wifi-reborn' {
 
     export function connectToSSID(SSID: string): Promise<void>;
     export function connectToSSIDPrefix(SSIDPrefix: string): Promise<void>;
+    export function connectToSSIDPrefix(SSIDPrefix: string, joinOnce: boolean): Promise<void>;
```

Comment:
There is no need to duplicate the function declaration. You can just change the function interface along these lines:
```ts
export function connectToSSIDPrefix(SSIDPrefix: string, joinOnce?: boolean)
```
The same with all the other ones.

Reply:
@Rapsssito thank you for the feedback! I have pushed updates removing the duplicate function declarations.

Probably outside the scope of this PR, but just a thought- would it be useful to provide a default value in javascript instead of overloading the method in `RNWifi.m`? It could make for a cleaner interface to the native module and reduce overall code.

ex.
```
RCT_EXPORT_METHOD(connectToProtectedSSID:(NSString*)ssid
                  withPassphrase:(NSString*)passphrase
                  isWEP:(BOOL)isWEP
                  joinOnce:(BOOL)joinOnce
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
```

and in `index.js`

```
export const connectToProtectedSSID = (SSID, password = '', isWEP = false, joinOnce = false) =>
    WifiManager.connectToProtectedSSID(SSID, password, isWEP, joinOnce);
```

(seen in https://github.com/facebook/react-native/issues/12498#issuecomment-281834100)

Reply:
You are right! I like your idea. But this must also be changed in Android.

Reply:
Yes that's true. Maybe I can address it in a follow up but it would take some time